src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Windows.cs

```diff
@@ -548,10 +549,31 @@ void Resize(Size newSize)
 			{
 				_currentSize = newSize;
 
-				UpdateItemsSource();
-				UpdateSnapPointsType();
-				UpdateSnapPointsAlignment();
-				UpdateCarouselViewInitialPosition();
+				if (_isCarouselViewReady)
+					InvalidateItemSize();
+				else
+					InitialSetup();
+
+				_isCarouselViewReady = true;
+			}
+		}
+
+		void InitialSetup()
+		{
+			UpdateItemsSource();
+			UpdateSnapPointsType();
+			UpdateSnapPointsAlignment();
+			UpdateCarouselViewInitialPosition();
+		}
+
+		void InvalidateItemSize()
+		{
+			foreach (var item in ListViewBase.GetChildren<ItemContentControl>())
+			{
+				item.ItemHeight = GetItemHeight();
```

Comment:
`GetItemHeight` and `GetItemWidth` should be moved out of the `foreach` since they only need calculation once.

Reply:
Good suggestion. Changed.